pkg/pillar/cmd/volumemgr/updatestatus.go

```diff
@@ -26,12 +26,21 @@ func doUpdateContentTree(ctx *volumemgrContext, status *types.ContentTreeStatus)
 	addedBlobs := []string{}
 
 	if status.State < types.VERIFIED {
-		if status.DatastoreType == "" {
+		if !status.AllDatastoresResolved {
 			log.Functionf("contentTreeStatus(%s) does not have a datastore type yet, deferring", status.ContentID)
 			return false, false
 		}
 
-		if status.IsOCIRegistry() {
+		if status.IsOCIRegistry {
+			if len(status.FallbackDatastoreIDsList) != 0 {
```

Comment:
@giggsoff This is the check, which (I assume) should help not to touch the sha-resolve mechanism in this patchset. (you refer here https://github.com/lf-edge/eve/pull/2983#discussion_r1058246497)

Reply:
Oh, I see, we have explicit check for that case, thank you!